src/unk_02054884.c

```diff
@@ -14,17 +14,14 @@
 #include "party.h"
 #include "overlay005/ov5_021E622C.h"
 
-BOOL sub_02054884 (Pokemon * param0)
+BOOL Pokemon_CanBattle (Pokemon *mon)
 {
-    if (Pokemon_GetValue(param0, MON_DATA_CURRENT_HP, NULL) == 0) {
-        return 0;
-    }
-
-    if (Pokemon_GetValue(param0, MON_DATA_IS_EGG, NULL)) {
-        return 0;
+    // this can be simplified further, but it won't match
+    if (Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) == 0) {
+        return FALSE;
```

Comment:
How did you try to simplify this?

Reply:
I tried these:
```
return !(Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) == 0) || !Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL);
```

```
return Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) != 0 || !Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL);
```
```
return !(Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) == 0 && Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL));
```

Reply:
I took a stab at this and got to this:

```c
BOOL Pokemon_CanBattle(void *mon)
{
    return Pokemon_GetValue(mon, MON_DATA_CURRENT_HP, NULL) != 0
        && !Pokemon_GetValue(mon, MON_DATA_IS_EGG, NULL);
}
```

which is logically (and functionally) equivalent, but does not match byte-for-byte. So, meh. Keep this as-is.